src/sqlfluff/core/rules/functional/segments.py

```diff
@@ -88,11 +88,19 @@ def raw_segments(self) -> "Segments":
             raw_segments_list.extend(s.raw_segments)
         return Segments(*raw_segments_list, templated_file=self.templated_file)
 
+    def recursive_crawl_all(self) -> "Segments":  # pragma: no cover
+        """Recursively crawl all descendant segments."""
+        segments: List[BaseSegment] = []
+        for s in self:
+            for i in s.recursive_crawl_all():
+                segments.append(i)
+        return Segments(*segments, templated_file=self.templated_file)
+
     def recursive_crawl(self, *seg_type: str, recurse_into: bool = True) -> "Segments":
         """Recursively crawl for segments of a given type."""
         segments: List[BaseSegment] = []
         for s in self:
-            for i in s.recursive_crawl(*seg_type, recurse_into):
+            for i in s.recursive_crawl(*seg_type, recurse_into=recurse_into):
```

Comment:
Do we need this change?

Reply:
Nope. Will remove.

Reply:
Hmm, mypy complains if I remove it. It thinks we're trying to pass `recurse_into` (a boolean value) as one of the type values. I think mypy is wrong, but I'm okay with this change -- passing it explicitly as a keyword argument _does_ make it a bit clearer, even if the old way works.

Reply:
I think I know why `mypy` started complaining about this. In this PR, I added type annotations to the `recursive_crawl()` function. Previously, `mypy` didn't know that the `recurse_into` parameter of `BaseSegment.recursive_crawl()` was a boolean. Now that it has that information, it started to complain.

Reply:
Fine keeping it in.